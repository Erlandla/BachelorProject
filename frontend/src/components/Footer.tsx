import React from 'react'

function Footer() {
  const backToTopfunction = () => {
    document.body.scrollTop = 0 // For Safari
    document.documentElement.scrollTop = 0 // for safari, chrome, edge, etc.
  }

  return (
    <div className="footerbox bg-footer h-20 place-items-center grid grid-cols-3 bottom-0 w-full">
      <div>
        <button
          className={
            'toTopButton bg-buttonDark hover:bg-buttonHover text-white rounded-lg w-32 sm:w-40 h-8'
          }
          onClick={backToTopfunction}
        >
          <p className="text-xs sm:text-base"> Tilbake til toppen </p>
        </button>
      </div>
      <p className="text-white text-xs sm:text-base"> Laget av snille gutter fra NTNU ♥️ </p>
    </div>
  )
}

export default Footer
