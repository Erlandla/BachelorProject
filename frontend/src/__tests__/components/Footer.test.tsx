import React from "react";
import { cleanup, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Footer from "../../components/Footer";
import renderer from "react-test-renderer";

afterEach(() => {
    cleanup();
})

it("Footer renders correctly", () => {
    render(<Footer />)
    // Note: test will fail if text gets changed
    expect(screen.getByText("Laget av snille gutter fra NTNU ♥️")).toBeInTheDocument();
});

it("'Tilbake til toppen'-button is clickable", () => {
    render(<Footer />)
    // Note: test will fail if text gets changed
    userEvent.click(screen.getByText("Tilbake til toppen"));
});

it("Matches snapshot", () => {
    const snapshot = renderer.create(<Footer/>).toJSON();
    expect(snapshot).toMatchSnapshot();
});