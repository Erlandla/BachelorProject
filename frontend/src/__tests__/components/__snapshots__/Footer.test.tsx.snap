// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div
  className="footerbox bg-footer h-20 place-items-center grid grid-cols-3 bottom-0 w-full"
>
  <div>
    <button
      className="toTopButton bg-buttonDark hover:bg-buttonHover text-white rounded-lg w-32 sm:w-40 h-8"
      onClick={[Function]}
    >
      <p
        className="text-xs sm:text-base"
      >
         Tilbake til toppen 
      </p>
    </button>
  </div>
  <p
    className="text-white text-xs sm:text-base"
  >
     Laget av snille gutter fra NTNU ♥️ 
  </p>
</div>
`;
